File: backend/src/main/java/com/backend/domain/community/entity/Comment.java
package com.backend.domain.community.entity;

import com.backend.domain.analysis.entity.AnalysisResult;
import com.backend.global.entity.BaseEntity;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.SQLDelete;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
// SQL 문 delete가 실행 될 때 -> 실행될 SQL문을 설정
@SQLDelete(sql = "UPDATE Comment SET deleted = true WHERE id = ?")
// 조회 시에 기본적으로 deleted = false인 것을만 조회하도록 설정 -> 관리자 조회 시에는 삭제 처리 된 것도 조회 필요
// @Where(clause="deleted = false")
@Builder
public class Comment extends BaseEntity {
    // 댓글 id
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    // 분석결과 id
    @ManyToOne(optional= false)
    @JoinColumn(name = "analysis_result_id",  nullable = false)
    private AnalysisResult analysisResult;

    // 회원 id
    @Column(name = "member_id", nullable = false)
    private Long memberId;

    // 댓글 내용
    @Column(columnDefinition = "TEXT", nullable = false)
    private String comment;

    // 삭제 여부
    @Column(nullable = false)
    private boolean deleted;

    public void updateComment(String newContent) {
        if (newContent == null || newContent.isBlank()) {
            throw new IllegalArgumentException("댓글 내용은 비어 있을 수 없습니다.");
        }
        this.comment = newContent;
    }
}

File: backend/src/main/java/com/backend/domain/community/entity/Comment.kt
package com.backend.domain.community.entity

import com.backend.domain.analysis.entity.AnalysisResult
import com.backend.global.entity.BaseEntity
import jakarta.persistence.*
import org.hibernate.annotations.SQLDelete

@Entity
@SQLDelete(sql = "UPDATE Comment SET deleted = true WHERE id = ?")
class Comment(
    // 댓글 id
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    // 분석결과 id
    @ManyToOne(optional = false)
    @JoinColumn(name = "analysis_result_id", nullable = false)
    val analysisResult: AnalysisResult,

    // 회원 id
    @Column(name = "member_id", nullable = false)
    val memberId: Long?,

    // 댓글 내용
    @Column(columnDefinition = "TEXT", nullable = false)
    var comment: String = "",

    // 삭제 여부
    @Column(nullable = false)
    var deleted: Boolean = false

) : BaseEntity() {
    companion object {
        @JvmStatic
        fun create(
            analysisResult: AnalysisResult,
            memberId: Long?,
            content: String,
            deleted: Boolean = false
        ): Comment {
            require(content.isNotBlank()) { "댓글 내용은 비어 있을 수 없습니다." }

            return Comment(
                analysisResult = analysisResult,
                memberId = memberId,
                comment = content,
                deleted = deleted
            )
        }
    }

    fun updateComment(newContent: String) {
        require(newContent.isNotBlank()) { "댓글 내용은 비어 있을 수 없습니다." }
        this.comment = newContent
    }
}